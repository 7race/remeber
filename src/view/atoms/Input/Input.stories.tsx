import { ComponentStory, ComponentMeta } from '@storybook/react';
import { Input } from './Input';

export default {
  title: 'Input',
  component: Input,
  argTypes: {
    type: {
      type: 'string',
      options: ['text', 'password'],
      control: { type: 'radio' },
    },
  },
} as ComponentMeta<typeof Input>;

const Template: ComponentStory<typeof Input> = (args) => <Input {...args} />;

export const Default = Template.bind({});
