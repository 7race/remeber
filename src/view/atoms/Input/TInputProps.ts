import { ChangeEventHandler } from 'react';

export type TInputProps = {
  placeholder?: string;
  type?: 'password';
  variant?: 'grey';
  value: string;
  onChange: ChangeEventHandler;
};
