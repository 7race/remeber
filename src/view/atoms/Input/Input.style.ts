import styled, { css } from 'styled-components';
import { TInputProps } from './TInputProps';

const variants = {
  grey: css`
    background-color: ${({ theme }) => theme.colors.lightGrey};
    border: none;
    &::placeholder {
      color: ${({ theme }) => theme.colors.grey};
    }
  `,
};

export const InputPassword = styled.div`
  position: relative;
  & img {
    position: absolute;
    top: 50%;
    transform: translateY(-50%);
    margin-left: -50px;
    cursor: pointer;
    width: 30px;
    height: 30px;
  }
`;

export const Input = styled.input<TInputProps>`
  font-size: 20px;
  padding: 15px;
  border-radius: 8px;
  outline: none;
  padding-left: 20px;

  ${({ variant }) => (variant ? variants[variant] : null)}
`;
