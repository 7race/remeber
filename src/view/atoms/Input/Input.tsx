import { useState } from 'react';
import { eyeSolid, eyeSlash } from '@static';
import { TInputProps } from './TInputProps';
import * as S from './Input.style';

const InputPassword = ({ placeholder, type, variant }: TInputProps) => {
  const [isPasswordVisible, setIsPasswordVisible] = useState<boolean>(false);

  const changeVisible = () => setIsPasswordVisible((prev) => !prev);
  return (
    <S.InputPassword>
      {isPasswordVisible ? (
        <>
          <S.Input placeholder={placeholder} variant={variant} />
          <img src={eyeSolid} alt='opened eye' onClick={changeVisible} />
        </>
      ) : (
        <>
          <S.Input placeholder={placeholder} type={type} variant={variant} />
          <img src={eyeSlash} alt='closed eye' onClick={changeVisible} />
        </>
      )}
    </S.InputPassword>
  );
};

export const Input = ({ placeholder, type, variant }: TInputProps) =>
  type === 'password' ? (
    <InputPassword placeholder={placeholder} type={type} variant={variant} />
  ) : (
    <S.Input placeholder={placeholder} type={type} variant={variant} />
  );
