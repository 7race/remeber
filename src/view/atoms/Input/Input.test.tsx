import userEvent from '@testing-library/user-event';
import { render, screen } from '@tests-utils';
import { Input } from './Input';

const onChange = jest.fn();

describe('input component', () => {
  it('correct styles for input type text', () => {
    const { container } = render(<Input value='' onChange={onChange} />);
    expect(container).toMatchSnapshot();
  });

  it('correct styles for input type password', () => {
    const { container } = render(<Input type='password' value='' onChange={onChange} />);
    expect(container).toMatchSnapshot();
  });

  it('hide/show password by click on eye icon', async () => {
    const user = userEvent.setup();
    const { container } = render(<Input type='password' value='' onChange={onChange} />);
    expect(container).toMatchSnapshot();
    const closedEye = screen.getByAltText('closed eye');
    await user.click(closedEye);
    expect(container).toMatchSnapshot();
  });

  it('correct styles of variant grey', () => {
    const { container } = render(<Input variant='grey' value='' onChange={onChange} />);
    expect(container).toMatchSnapshot();
  });
});
