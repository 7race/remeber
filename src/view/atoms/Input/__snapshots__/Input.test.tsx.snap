// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input component correct styles for input type password 1`] = `
<div>
  <div
    class="sc-bczRLJ dkamqk"
  >
    <input
      class="sc-gsnTZi hsxpLY"
      type="password"
      value=""
    />
    <img
      alt="closed eye"
      src="eye-slash.svg"
    />
  </div>
</div>
`;

exports[`input component correct styles for input type text 1`] = `
<div>
  <input
    class="sc-gsnTZi hsxpLY"
    value=""
  />
</div>
`;

exports[`input component correct styles of variant grey 1`] = `
<div>
  <input
    class="sc-gsnTZi fqhQFY"
    value=""
  />
</div>
`;

exports[`input component hide/show password by click on eye icon 1`] = `
<div>
  <div
    class="sc-bczRLJ dkamqk"
  >
    <input
      class="sc-gsnTZi hsxpLY"
      type="password"
      value=""
    />
    <img
      alt="closed eye"
      src="eye-slash.svg"
    />
  </div>
</div>
`;

exports[`input component hide/show password by click on eye icon 2`] = `
<div>
  <div
    class="sc-bczRLJ dkamqk"
  >
    <input
      class="sc-gsnTZi hsxpLY"
      value=""
    />
    <img
      alt="opened eye"
      src="eye.svg"
    />
  </div>
</div>
`;
